pydra/engine/tests/test_helpers.py

```diff
@@ -135,7 +135,9 @@ def test_hash_value_dir(tmpdir):
     with open(file_2, "w") as f:
         f.write("hi")
 
-    assert hash_value(tmpdir, tp=Directory) == hash_value([file_1, file_2], tp=File)
+    assert sorted(hash_value(tmpdir, tp=Directory)) == sorted(
```

Comment:
@kaczmarj - i think it should be the other way around. i.e. the change should be in the function to always return a consistent hash, and the sort could be on the returned hashes, but more importantly a directory hash should be a singleton (i.e a str) not a list. 

so this might be better addressed by fixing the hash function.

Reply:
ok. at the moment, `hash_value(tmpdir, tp=Directory)` returns one hash per file in the directory. you would prefer that to be a single string, representing a hash of all the files in the directory, correct?

Reply:
yes, it should also support nested directories. this could be easily done by generating a list of hashes of a sorted directory tree (via os.walk and perhaps ignoring empty directories and hidden files), and then hashing that list.

Reply:
this function does support nested directory - see https://github.com/nipype/pydra/blob/master/pydra/engine/tests/test_helpers.py#L142

Indeed it doesn't return one value (but I thought that the order is always the same...), the `hash` function from the input spec returns one value.

This can be changed to one value, but not sure if it changes anything, the order has to be assured anyway

Reply:
@kaczmarj - do you want me to finish this?

Reply:
@satra @djarecka - please see my updates. the `hash_dir` function now returns a single hash value and the directory tree is sorted